code/game/objects/effects/effect_system.dm

```diff
@@ -155,10 +155,6 @@ steam.start() -- spawns the effect
 				for(i=0, i<pick(1,2,3), i++)
 					sleep(5)
 					step(sparks,direction)
-				spawn(20)
-					if(sparks)
-						qdel(sparks)
-					src.total_sparks--

```

Comment:
You might want to throw this up on the spawn(20), in-case it does something important.